src/forms/Typeahead.jsx

```diff
@@ -12,6 +12,44 @@ export const TA_ITEM_CLASSNAME = 'typeahead-item';
  * @module Typeahead
  */
 class Typeahead extends React.PureComponent {
+	constructor(props) {
+		super(props);
+		this.onItemMouseEnter = this.onItemMouseEnter.bind(this);
+		this.onItemKeyUp = this.onItemKeyUp.bind(this);
+
+		this.state = {
+			actualHighlightedIndex: 0
```

Comment:
no harm in just calling this `highlightedIndex`, since this internal state is now the point of truth for which item is highlighted